README.md

```diff
@@ -48,22 +93,39 @@ if let foo = foo {
 }
 ```
 
-Alternatively, you might want to use Swift's Optional Chaining in some of these cases, such as:
+When dealing with `weak` or optional delegates, optional binding syntax is always preferable unless functionality is dependent on that value being unwrapped.
```

Comment:
I think my main preference here is that it's more compact.  I was trying to leave an opening above for situations where functionality is dependent on the object existing.  Here's the case I'm thinking of:

```
self?.delegate?.somethingHappened(self)
```

Won't work since the argument is dependent on a non-optional version of `self`.  So you'd have to do:

```
if let self = self, let delegate = delegate {
    delegate.somethingHappened(self)
}
```

Any ideas how to state this more clearly?